src/popup/components/Modals/ConfirmTransactionSign.vue

```diff
@@ -153,153 +183,190 @@ export default {
-          bytecode,
-          calldata: this.transaction.callData,
-        });
-        this.txFunction = txParams.function;
-        const allTokens = this.getTokens(txParams);
-
-        this.tokenList = allTokens.map((token) => ({
-          ...token,
-          tokens: token.isPool && !this.isProvideLiquidity
-            ? allTokens.filter((t) => !t.isPool).reverse()
-            : [token],
-        }));
-      } catch (e) {
-        this.tokenList = null;
-        this.txFunction = null;
-      } finally {
-        this.loading = false;
+  setup(props, { root }) {
+    const { getSdk } = useSdk({ store: root.$store });
+
+    const showAdvanced = ref(false);
+    const tokenList = ref<ITokenResolved[]>([]);
+    const txFunction = ref<TxFunctionRaw | undefined>();
+    const loading = ref(false);
+
+    const availableTokens = useState('fungibleTokens', 'availableTokens');
+    const getTxSymbol = useGetter('getTxSymbol');
+    const getTxAmountTotal = useGetter('getTxAmountTotal');
+    const getTxFee = useGetter('getTxFee');
+    const getNameFee = useGetter('getNameFee');
+    const getDexContracts = useGetter('getDexContracts');
+    const getTxDirection = useGetter('getTxDirection');
+    const isTxAex9 = useGetter('isTxAex9');
+
+    const isAllowance = computed(() => (
+      txFunction.value
+      && FUNCTION_TYPE_DEX.allowance.includes(txFunction.value)
+    ));
+
+    const isSwap = computed(
+      () => txFunction.value && FUNCTION_TYPE_DEX.swap.includes(txFunction.value),
+    );
+
+    const isPool = computed(
+      () => txFunction.value && FUNCTION_TYPE_DEX.pool.includes(txFunction.value),
+    );
+
+    const isDex = computed(() => (
+      getDexContracts.value.router.includes(props.transaction?.contractId)
+      || getDexContracts.value.wae.includes(props.transaction?.contractId)
+      || isAllowance.value
+    ));
+
+    const swapDirection = computed(() => {
+      if (txFunction.value) {
+        if (FUNCTION_TYPE_DEX.maxSpent.includes(txFunction.value)) {
+          return 'maxSpent';
+        }
+        if (FUNCTION_TYPE_DEX.minReceived.includes(txFunction.value)) {
+          return 'minReceived';
+        }
       }
-    }
-  },
-  methods: {
-    convertToken,
-    getTokens(txParams) {
-      if (!this.isDex) return [this.singleToken];
-      const resolver = transactionTokenInfoResolvers[camelCase(txParams.function)];
+      return 'total';
+    });
+
+    const singleToken = computed((): ITokenResolved => ({
+      isReceived: getTxDirection.value(props.transaction) === TX_FUNCTIONS.received,
+      amount: getTxAmountTotal.value(props.transaction),
+      symbol: getTxSymbol.value(props.transaction),
+    }));
+
+    const filteredTxFields = computed(
+      () => TX_FIELDS_TO_DISPLAY.filter((field) => !!props.transaction[field]),
+    );
+
+    const swapTokenAmountData = computed(
+      (): ITokenResolved => swapDirection.value === 'maxSpent' ? tokenList.value[0] : tokenList.value[1],
```

Comment:
'maxSpent' -> constant?

Reply:
I'm leaving this for now as this is a key used to grab translation. I need to dive deeper with this, like for example get rid of parsing of the translation paths.